src/Microsoft.ML.Transforms/FeatureContributionCalculationTransform.cs

```diff
@@ -0,0 +1,370 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Transforms;
+
+[assembly: LoadableClass(FeatureContributionCalculatingTransformer.Summary, typeof(FeatureContributionCalculatingTransformer), null, typeof(SignatureLoadModel),
+    FeatureContributionCalculatingTransformer.FriendlyName, FeatureContributionCalculatingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(FeatureContributionCalculatingTransformer), null, typeof(SignatureLoadRowMapper),
+   FeatureContributionCalculatingTransformer.FriendlyName, FeatureContributionCalculatingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(void), typeof(FeatureContributionEntryPoint), null, typeof(SignatureEntryPointModule), FeatureContributionCalculatingTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// The FeatureContributionCalculationTransformer scores the model on an input dataset and
+    /// computes model-specific contribution scores for each feature. See the list of currently supported predictors below.
+    /// </summary>
+    /// <remarks>
+    /// Feature Contribution Calculation is currently supported for the following Predictors:
+    ///     Regression:
+    ///         OrdinaryLeastSquares, StochasticDualCoordinateAscent (SDCA), OnlineGradientDescent, PoissonRegression,
+    ///         GeneralizedAdditiveModels (GAM), LightGbm, FastTree, FastForest, FastTreeTweedie
+    ///     Binary Classification:
+    ///         AveragedPerceptron, LinearSupportVectorMachines, LogisticRegression, StochasticDualCoordinateAscent (SDCA),
+    ///         StochasticGradientDescent (SGD), SymbolicStochasticGradientDescent, GeneralizedAdditiveModels (GAM),
+    ///         FastForest, FastTree, LightGbm
+    ///     Ranking:
+    ///         FastTree, LightGbm
+    ///
+    /// See the sample below for an example of how to compute feature importance using the FeatureContributionCalculatingTransformer.
+    /// </remarks>
+    /// <example>
+    /// <format type="text/markdown">
+    /// <![CDATA[
+    /// [!code-csharp[FCT](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/FeatureContributionCalculationTransform.cs)]
+    /// ]]>
+    /// </format>
+    /// </example>
+    public sealed class FeatureContributionCalculatingTransformer : RowToRowTransformerBase
```

Comment:
>FeatureContributionCalculatingTransformer [](start = 24, length = 41)

So I might have expected this to live in the same assembly as where the scorer lives. Why is it way off in `Microsoft.ML.Transforms`? Does it use something in this assembly that required this separation?

Reply:
Since this is a transform, I thought it was better to have it in the Transforms assembly. I can put it back in data though. Maybe under the transforms folder?

---
In reply to: [243037354](https://github.com/dotnet/machinelearning/pull/1847#discussion_r243037354) [](ancestors = 243037354)